src/components/Article.vue

```diff
@@ -0,0 +1,291 @@
+            class="video"
+            width="100%"
+            height="50%"
+            :src="'https://www.youtube.com/embed/' + apiData.video.video_id"
+        />
+        <router-link :to="{ name: modalRoute, params: { id: apiData.id } }" class="text">
+            <h2 v-if="apiData.title !== ''">
+                {{ apiData.title }}
+            </h2>
+            <b v-if="apiData.section !== ''">{{ apiData.section }}</b>
+            <p>{{ apiData.text | striphtml }}</p>
+        </router-link>
+        <footer>
+            <p class="time">
+                {{ apiData.published | moment }}
+            </p>
+            <br />
+            <p>
+                {{ apiData.author.name }}
+            </p>
+            <div v-if="showMsg === true" class="msg">
+                <p>Favorites are only stored locally on this device!</p>
+                <div class="arrow-down" />
+            </div>
+            <img
+                v-if="favorite === false"
+                class="favoriteIcon"
+                src="/src/assets/icons/baseline-favorite-border.svg"
+                alt="Add to favorites."
+                @click="addFavorite()"
+            />
+            <img
+                v-if="favorite === true"
+                class="favoriteIcon"
+                src="/src/assets/icons/baseline-favorite.svg"
+                alt="Remove from favorites."
+                @click="removeFavorite()"
+            />
+        </footer>
+    </div>
+</template>
+
+<script>
+export default {
+    props: {
+        apiData: {
+            type: Object,
+            default() {
+                return { message: 'Api Data' };
+            },
+        },
+        modalRoute: {
+            type: String,
+            default: 'modalStart',
+        },
+    },
+    data() {
+        return {
+            slideIndex: 1,
+            favorite: false,
+            showMsg: false,
+            time: '',
+        };
+    },
+    mounted() {
+        this.imageSlider(this.slideIndex);
+        /* load all favorites as an array and loop thru it comparing ids
+        if they match the heart gets a different icon */
+        const data = JSON.parse(localStorage.getItem('id'));
+
+        data.forEach(favorit => {
+            if (this.apiData.id === favorit) {
+                this.favorite = true;
+            }
+        });
+        this.time = new Date();
+    },
+    methods: {
+        /* next image in slider */
+        nextImage(n) {
+            this.imageSlider((this.slideIndex += n));
+        },
+
+        /* if the article contains more than one picture it shows a slider */
+        imageSlider(n) {
+            let i;
+            const x = document.getElementsByClassName(`${this.apiData.id}Image`);
+            if (n > x.length) {
+                this.slideIndex = 1;
+            }
+            if (n < 1) {
+                this.slideIndex = x.length;
+            }
+
+            for (i = 0; i < x.length; i += 1) {
```

Comment:
Variabeln i används bara innanför for loopen. Definera därför `i` i loopen. Ta sedan bort rad 115 med `let i;`
```js
for (let i = 0; i < x.length; i += 1) 
```